en/05_Contributing/04_Release_Process.md

```diff
@@ -150,59 +150,88 @@ In these cases, it is best practice to "refactor" these APIs into something more
+* Deprecation::notice('1.2.3', 'Use anotherMethod() instead'); to the top of the method
+
+When deprecating a class:
+
+* Add the following docblock `@deprecated 1.2.3 Use AnotherClass instead`
+* Add `Deprecation::notice('1.2.3', 'Use AnotherClass instead', SCOPE_CLASS);` to the top of `__construct()`
+
+When deprecating config:
+
+* Add the following docblock `@deprecated 1.2.3 Use different_config instead`
+
+If there is no immediate replacement for deprecated code that is being called, either because the replacement is not available until the next major version, or because there is not a plan for there to be a replacement, the message should be "Will be removed without equivalent functionality to replace it."
+
+ ### Avoiding deprecated API
+ 
+ Wherever possible, once some API has been deprecated, we should stop using it. This allows projects to emit deprecation notices and have an accurate list of code that they need to change.
+ 
+ In some cases, we may not yet have a replacement for code that is required for the current major - but that we still have to call and support internally. In these cases, since we can't stop calling the code, wrap the call to the deprecated code in `Deprecation::withNoReplacement()` e.g:
+
+```php
+Deprecation::withNoReplacement(function () {
+  $this->myDeprecatedMethod();
+});
+```
+
+For any unit tests using the deprecated method/class/config, add the following the the top of the unit test. This ensures that deprecated code is still supported as usual when `Deprecation` notices are not enabled, though the tests are skipped whenever you are testing to see if deprecated code is still be called.
+```php
+if (Deprecation::isEnabled()) {
+    $this->markTestSkipped('Test calls deprecated code');
+}
+```
 
 Here's an example for replacing `Director::isDev()` with a (theoretical) `Env::is_dev()`:
 
 ```php
 /**
- * Returns true if you are in development mode
- * @deprecated 4.0 Use {@link Env::is_dev()} instead.
+ * Returns true if your are in development mode
+ * @deprecated 4.12.0 Use Env::is_dev() instead.
  */
 public function isDev() 
 {
-    Deprecation::notice('4.0', 'Use Env::is_dev() instead');
+    Deprecation::notice('4.12.0', 'Use Env::is_dev() instead');
     return Env::is_dev();
 }
 ```
 
-This change could be committed to a minor release like *3.2.0*, and remains deprecated in all subsequent minor releases
-(e.g. *3.3.0*, *3.4.0*), until a new major release (e.g. *4.0.0*), at which point it gets removed from the codebase. 
-
-Deprecation notices are enabled by default on dev environment, but can be
-turned off via either `.env` or in your _config.php. Deprecation
-notices are always disabled on both live and test.
+This change could be committed to a minor release like *4.12.0*, and remains deprecated in all subsequent minor releases
+(e.g. *4.13.0*), until a new major release (e.g. *5.0.0*), at which point it gets removed from the codebase. 
 
+## Enabling deprecation notices
```

Comment:
```suggestion
### Enabling deprecation notices
```
Level three heading, so it sits under the general "Deprecation" section.

Reply:
Updated